packages/mui-system/src/createStyled.js

```diff
@@ -137,8 +138,16 @@ export default function createStyled(input = {}) {
           if (styleOverrides) {
             const resolvedStyleOverrides = {};
             Object.entries(styleOverrides).forEach(([slotKey, slotStyle]) => {
-              resolvedStyleOverrides[slotKey] =
-                typeof slotStyle === 'function' ? slotStyle(props) : slotStyle;
+              const slotValue = typeof slotStyle === 'function' ? slotStyle(props) : slotStyle;
+              if (typeof slotStyle === 'function' && typeof slotValue === 'string') {
+                throw new MuiError(
+                  "MUI: The return type of the callback in '%s.styleOverrides.%s' cannot be a string. Change the return type to array or object instead.",
```

Comment:
What's the link with a blog post? Could we move it to a different PR?

Reply:
Fine for me.